### PR TITLE
Expose OrchestratorServiceStubs

### LINKED ISSUE #882: Allow to register new search attributes
**Is your feature request related to a problem? Please describe.**
Temporal SDK allows to communicate with temporal server through GRPC calls. Currently you can create workflows with custom search attributes but those attributes must be already registered on Elasticsearch through the `add-search-attributes` sub-command on `tctl`

**Describe the solution you'd like**
Allow to register new search attributes through the SDK

**Describe alternatives you've considered**
The only alternative by now is to run tctl 

**Additional context**
Nope

### PR DESCRIPTION
**Pull request by Spikhalskiy**
## What was changed
WorkflowServiceStubsImpl was refactored to reduce code duplication

## Why?
Allow users to register search attributes using sdk
Refactoring of WorkflowServiceStubs to reduce code duplication

Closes #882

### CODE REVIEW

File: temporal-serviceclient/src/main/java/io/temporal/serviceclient/OperatorServiceStubs.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import io.temporal.api.operatorservice.v1.OperatorServiceGrpc;
+import io.temporal.internal.WorkflowThreadMarker;
+
+/** Initializes and holds gRPC blocking and future stubs. */
+public interface OperatorServiceStubs
+    extends ServiceStubs<
+        OperatorServiceGrpc.OperatorServiceBlockingStub,
+        OperatorServiceGrpc.OperatorServiceFutureStub> {
+  String HEALTH_CHECK_SERVICE_NAME = "temporal.api.workflowservice.v1.WorkflowService";
```
Comment (cretz) [inline temporal-serviceclient/src/main/java/io/temporal/serviceclient/OperatorServiceStubs.java:32]: Should we change this service name?

File: temporal-serviceclient/src/main/java/io/temporal/serviceclient/ChannelManager.java [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+  private void initConnectionManagement() {
+    // Currently, it is impossible to modify backoff policy on NettyChannelBuilder.
+    // For this reason we reset connection backoff every few seconds in order to limit maximum
+    // retry interval, which by default equals to 2 minutes.
+    // Once https://github.com/grpc/grpc-java/issues/7456 is done we should be able to define
+    // custom policy during channel creation and get rid of the code below.
+    if (options.getConnectionBackoffResetFrequency() != null) {
+      grpcConnectionManager.scheduleWithFixedDelay(
+          resetGrpcConnectionBackoffTask(),
+          options.getConnectionBackoffResetFrequency().getSeconds(),
```
Comment (cretz) [inline temporal-serviceclient/src/main/java/io/temporal/serviceclient/ChannelManager.java:195]: There is a lot of `getSeconds()` here and health check timeout and such. Can we use the more granular unit, e.g. milliseconds?

File: temporal-serviceclient/src/main/java/io/temporal/serviceclient/ChannelManager.java [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+      ServiceStubsOptions options, List<ClientInterceptor> additionalHeadInterceptors) {
+    // Do not shutdown a channel passed to the constructor from outside
+    this.channelNeedsShutdown = options.getChannel() == null;
+
+    this.options = options;
+    if (options.getChannel() != null) {
+      this.channel = options.getChannel();
+      this.grpcConnectionManager = null;
+    } else {
+      this.channel = prepareChannel();
```
Comment (cretz) [inline temporal-serviceclient/src/main/java/io/temporal/serviceclient/ChannelManager.java:79]: To confirm, by default we are intentionally creating separate channels for different gRPC services on the same server? I haven't put in thought to how the API would look if you wanted to share them, but I just want to make sure that is intentional.

File: temporal-serviceclient/src/main/java/io/temporal/serviceclient/OperatorServiceStubsImpl.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        new ChannelManager(this.options, Collections.singletonList(deadlineInterceptor));
+
+    HealthCheckResponse healthCheckResponse =
+        this.channelManager.waitForServer(HEALTH_CHECK_SERVICE_NAME);
+    if (!HealthCheckResponse.ServingStatus.SERVING.equals(healthCheckResponse.getStatus())) {
+      throw new RuntimeException(
+          "Health check returned unhealthy status: " + healthCheckResponse.getStatus());
+    }
+
+    log.info(String.format("Created WorkflowServiceStubs for channel: %s", channelManager.channel));
```
Comment (cretz) [inline temporal-serviceclient/src/main/java/io/temporal/serviceclient/OperatorServiceStubsImpl.java:60]: Nit - I am used to seeing something like `log.info("Created WorkflowServiceStubs for channel: {}", channelManager.channel);` so the channel's `toString` is only invoked if that log level is enabled.

File: temporal-serviceclient/src/main/java/io/temporal/serviceclient/OperatorServiceStubs.java (hunk 1 above)
Comment (Spikhalskiy) [reply temporal-serviceclient/src/main/java/io/temporal/serviceclient/OperatorServiceStubs.java:32]: No, there is nothing responding to the OperatorService yet. This is as good as it gets for right now.
https://github.com/temporalio/temporal/pull/2579

File: temporal-serviceclient/src/main/java/io/temporal/serviceclient/OperatorServiceStubsImpl.java (hunk 4 above)
Comment (Spikhalskiy) [reply temporal-serviceclient/src/main/java/io/temporal/serviceclient/OperatorServiceStubsImpl.java:60]: It's a relocated code, but you are right, it's better to rework.

File: temporal-serviceclient/src/main/java/io/temporal/serviceclient/ChannelManager.java (hunk 3 above)
Comment (Spikhalskiy) [reply temporal-serviceclient/src/main/java/io/temporal/serviceclient/ChannelManager.java]: Yes. By default, every stub will have its own channel. It happens now if we create several WorkflowServiceStubs too. At the same time, if the users really want, they can reuse the channel. I exposed ServiceStubs#getRawChannel and the channel can be taken from the existing stubs and reused in the Options for a new stub.

Also, keep in mind that a regular App shouldn't be creating OperatorServiceStubs at all.

**Review by cretz — APPROVED** [review]

**Pull request merged**